src/dso.ts

```diff
@@ -52,23 +103,54 @@ export const dso = {
     return model;
   },
 
+  /*
+  * Transaction object selected for each driver 
+  */
   transaction: Transaction.transaction,
 
   /**
-   * connect to database
+   * connect to database mysql | postgres | Sqlite
    * @param config client config
    */
-  async connect(config: ClientConfig | Client) {
-    if (config instanceof Client) {
-      _client = config;
-    } else {
-      _client = new Client();
-      await _client.connect(config);
+  async connect<T extends PostgresConfig | MysqlConfig | SqliteConfig>(
+    config: T,
+  ): Promise<_ClientType | undefined> {
+    if (config["type"].toUpperCase() === "POSTGRES") {
+      _clientPostgres = new PostgresClient(config["clientConfig"]);
+      await _clientPostgres.connect();
+
+      return _configClientReturn = {
+        postgres: _clientPostgres,
+      };
+    } else if (config["type"].toUpperCase() === "MYSQL") {
+      if (config["client"] && config["client"] instanceof Client) {
+        _client = config["client"];
+      } else {
+        _client = new Client();
+        await _client.connect(config["clientConfig"]);
+      }
+      return _configClientReturn = {
+        mysql: _client,
+      };
+    } else if (config["type"].toUpperCase() === "SQLITE") {
+      const configgy: any = config["clientConfig"];
+
+      _clientSqlite = new SqliteClient(configgy["database"]);
+
+      return _configClientReturn = {
+        sqlite: _clientSqlite,
+      };
```

Comment:
Since different databases will inherit and implement a unified DsoClient abstract class, I think such a large amount of template code will be avoided

Reply:
I will update with the changes requested.